File: Kaskus/GridViewPaging/GridViewPagingCs/Datas/RepositoryDataTable.cs
namespace Dede.App.GridViewPaging.Datas
{
    using System;
    using System.Data;
    using System.Globalization;
    using System.Linq;

    using Dede.App.GridViewPaging.Pages;

    public sealed class RepositoryDataTable
        : IDataSourceRepository
    {
        private static DataTable gridDataTable = new DataTable();

        public DataSourceResult GetData(
            Int32 pageSize,
            Int32 currentPage)
        {
            InitGridDataTable();

            return new DataTableResult
            {
                Records = gridDataTable
                    .AsEnumerable()
                    .Skip(currentPage)
                    .Take(pageSize)
                    .CopyToDataTable(),
                TotalRecord = gridDataTable.Rows.Count
            };
        }

        private static void InitGridDataTable()
        {
            if (!gridDataTable.Rows.Count.Equals(0))
            {
                return;
            }

            gridDataTable.Rows.Clear();
            gridDataTable = GenerateGridDataTable();
        }

        private static DataTable GenerateGridDataTable()
        {
            var rnd = new Random(100000);
            var dat = DateTime.Now;

            var dt = new DataTable();

            dt.Columns.Add(@"Id", typeof(Guid));
            dt.Columns.Add(@"ItemString", typeof(String));
            dt.Columns.Add(@"ItemInt32", typeof(Int32));
            dt.Columns.Add(@"ItemBoolean", typeof(Boolean));
            dt.Columns.Add(@"ItemDecimal", typeof(Decimal));
            dt.Columns.Add(@"ItemDateTime", typeof(DateTime));

            for (var i = 0; i < 500; i++)
            {
                var itemRandom = rnd.Next();

                var mod = itemRandom % 2;
                var dec = itemRandom * 0.2M;
                dat = dat.AddDays(0 - (itemRandom / (Int32)dec));

                dt.Rows.Add(
                    Guid.NewGuid(),
                    itemRandom.ToString(CultureInfo.InvariantCulture),
                    i,
                    mod == 0,
                    dec,
                    dat);
            }

            return dt;
        }
    }
}

File: Kaskus/GridViewPaging/GridViewPagingVb/Datas/RepositoryDataTable.vb
Imports System
Imports System.Data
Imports System.Globalization
Imports Dede.App.GridViewPaging.Pages

Namespace Datas

    Public NotInheritable Class RepositoryDataTable
        Implements IDataSourceRepository

        Private Shared _gridDataTable As New DataTable()

        Public Function GetData(pageSize As Int32, currentPage As Int32) _
            As DataSourceResult _
            Implements IDataSourceRepository.GetData

            InitGridDataTable()

            Return New DataTableResult() With { _
                .Records = _gridDataTable.AsEnumerable().Skip(currentPage).Take(pageSize).CopyToDataTable(), _
                .TotalRecord = _gridDataTable.Rows.Count _
                }
        End Function

        Private Shared Sub InitGridDataTable()
            If Not _gridDataTable.Rows.Count.Equals(0) Then
                Return
            End If

            _gridDataTable.Rows.Clear()
            _gridDataTable = GenerateGridDataTable()
        End Sub

        Private Shared Function GenerateGridDataTable() As DataTable
            Dim rnd = New Random(100000)
            Dim dat = DateTime.Now

            Dim dt = New DataTable()
            dt.Columns.Add("Id", GetType(Guid))
            dt.Columns.Add("ItemString", GetType([String]))
            dt.Columns.Add("ItemInt32", GetType(Int32))
            dt.Columns.Add("ItemBoolean", GetType([Boolean]))
            dt.Columns.Add("ItemDecimal", GetType([Decimal]))
            dt.Columns.Add("ItemDateTime", GetType(DateTime))

            For i As Int32 = 0 To 800 - 1
                Dim itemRandom = rnd.Next()
                Dim modulus = itemRandom Mod 2
                Dim dec = itemRandom * 0.2D
                dat = dat.AddDays(0 - (itemRandom / CType(dec, Int32)))

                dt.Rows.Add(Guid.NewGuid(), itemRandom.ToString(CultureInfo.InvariantCulture), i, modulus = 0, dec, dat)
            Next

            Return dt
        End Function
    End Class
End Namespace